les/vflux/server/clientpool.go

```diff
@@ -0,0 +1,375 @@
+		if newValue != nil {
+			ns.SetStateSub(node, ppSetup.inactiveFlag, nodestate.Flags{}, 0)
+			cp.lock.RLock()
+			newValue.(*nodeBalance).SetPriceFactors(cp.defaultPosFactors, cp.defaultNegFactors)
+			cp.lock.RUnlock()
+		}
+	})
+
+	ns.SubscribeField(ppSetup.capacityField, func(node *enode.Node, state nodestate.Flags, oldValue, newValue interface{}) {
+		if c, ok := ns.GetField(node, clientField).(clientPeer); ok {
+			newCap, _ := newValue.(uint64)
+			c.UpdateCapacity(newCap, node == cp.capReqNode)
+		}
+	})
+	return cp
+}
+
+// AddMetrics adds metrics to the client pool. Should be called before Start().
+func (cp *ClientPool) AddMetrics(totalConnectedGauge metrics.Gauge,
+	clientConnectedMeter, clientDisconnectedMeter, clientActivatedMeter, clientDeactivatedMeter,
+	capacityQueryZeroMeter, capacityQueryNonZeroMeter metrics.Meter) {
+
+	cp.ns.SubscribeState(nodestate.MergeFlags(ppSetup.activeFlag, ppSetup.inactiveFlag), func(node *enode.Node, oldState, newState nodestate.Flags) {
+		if oldState.IsEmpty() && !newState.IsEmpty() {
+			clientConnectedMeter.Mark(1)
+		}
+		if !oldState.IsEmpty() && newState.IsEmpty() {
+			clientDisconnectedMeter.Mark(1)
+		}
+		if oldState.HasNone(ppSetup.activeFlag) && oldState.HasAll(ppSetup.activeFlag) {
+			clientActivatedMeter.Mark(1)
+		}
+		if oldState.HasAll(ppSetup.activeFlag) && oldState.HasNone(ppSetup.activeFlag) {
+			clientDeactivatedMeter.Mark(1)
+		}
+		_, connected := cp.Active()
+		totalConnectedGauge.Update(int64(connected))
+	})
+	cp.capacityQueryZeroMeter = capacityQueryZeroMeter
+	cp.capacityQueryNonZeroMeter = capacityQueryNonZeroMeter
+}
+
+// Start starts the client pool. Should be called before Register/Unregister.
+func (cp *ClientPool) Start() {
+	cp.ns.Start()
+}
+
+// Stop shuts the client pool down. The clientPeer interface callbacks will not be called
+// after Stop. Register calls will return nil.
+func (cp *ClientPool) Stop() {
+	cp.balanceTracker.stop()
+	cp.ns.Stop()
+}
+
+// Register registers the peer into the client pool. If the peer has insufficient
+// priority and remains inactive for longer than the allowed timeout then it will be
+// disconnected by calling the Disconnect function of the clientPeer interface.
+func (cp *ClientPool) Register(peer clientPeer) ConnectedBalance {
+	cp.ns.SetField(peer.Node(), clientField, clientPeerInstance{peer})
+	balance, _ := cp.ns.GetField(peer.Node(), btSetup.balanceField).(*nodeBalance)
+	return balance
+}
+
+// Unregister removes the peer from the client pool
+func (cp *ClientPool) Unregister(peer clientPeer) {
+	cp.ns.SetField(peer.Node(), clientField, nil)
+}
+
+// SetDefaultFactors sets the default price factors applied to subsequently connected clients
+func (cp *ClientPool) SetDefaultFactors(posFactors, negFactors PriceFactors) {
+	cp.lock.Lock()
+	cp.defaultPosFactors = posFactors
+	cp.defaultNegFactors = negFactors
```

Comment:
The price policies can be propagated to the balance tracker. So that the newly created balance can be assigned directly.

Reply:
done